setup.md

```diff
@@ -0,0 +1,92 @@
+### Install mbed-cli to a venv
+
+Install mbed-cli to a virtual environment, so that all the things mbed-cli
+installs automatically don't interfere with other things you have installed as
+part of your main environment (which may be managed by your package manager,
+etc.).
+
+```sh
+virtualenv venvs/mbed
+source ~/venvs/mbed/bin/activate
+pip install mbed-cli
+```
+
+### Install the GCC_ARM toolchain
```

Comment:
Is it really useful to have instructions for installing gcc-arm here, rather than linking to some generic instructions? For example, I wouldn't do any of this because I'm running Ubuntu and I instead do `apt-get install gcc-arm-none-eabi`.

Reply:
The Ubuntu supplied version may be out of date and not support ARMv8-M, for instance.

Reply:
Updated to link to generic download page and external instructions.